upgrade-steps.md

```diff
@@ -70,4 +70,51 @@ This documents the steps taken to upgrade VPB from ruby 1.93/rails 3.2 to the la
 |app/assets/javascripts/application.js|0%|0% (not needed for rails/ruby upgrade)|
 |app/assets/javascripts/aws-upload.js|0%|0% (not needed for rails/ruby upgrade)|
 
+## Dependency Versions
 
+|gem                      |Environment|Latest |Latest Ruby|Initial 3|Final 3  |Audited/Keep|Audit Notes|
+|-------------------------|-----------|-------|-----------|---------|---------|------------|-----------|
+|capistrano               |all        |3.12.1 |>= 2.0     |2.14.1   |*removed*|---         |Already removed |
+|capistrano-maintenance   |all        |1.2.1  |>= 0       |0.0.2    |*removed*|---         |Already removed |
+|sass-rails               |assets     |6.0.0  |>= 0       |3.2.6    |3.2.6    |Y/N         |No sass files present |
+|coffee-rails             |assets     |5.0.0  |>= 0       |3.2.2    |3.2.2    |Y/N         |No coffee files present |
+|therubyracer             |assets     |0.12.3 |>= 0       |0.12.1   |0.12.1   |Y/N?        |git commmit says its for asset precompile |
```

Comment:
This is a javascript engine for ruby. If coffee-rails is removed, and there isn't any javascript processing then this can probably be removed. Best way to test is to remove it and then do a production asset build.  If it doesn't break anything, you might just upgrade things and I'd bet that you'll get a deprecation warning by the rails build system.

Because of issues like this, it would be good if travis would try to do a production asset build as part of the build process. If configured right, this would catch issues in the asset build that are problematic for heroku and docker deploys. Since they both don't have access to the database when doing the build, and that was something that causes problems in the portal.

The tricky thing, that I remember is that the cached assets need to be cleared. And these cached assets can sometimes mess up development and testing. So it is probably best to do this after running all of the tests, and to make sure to clear the assets output folder (where ever that is), before running it. Travis might try to save the files that get stored in this folder to speed up future builds.